i18n/checkLocaleKeys.js

```diff
@@ -0,0 +1,90 @@
+import fs from 'fs'
+import path from 'path'
+import { fileURLToPath } from 'url'
+
+// Convert import.meta.url to __dirname equivalent
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+
+// Define the path to our i18n locales directory
+const localesI18nDirectory = path.join(__dirname, '/locales')
+
+// Import the key-checking function
+const checkLocaleKeys = () => {
+    const localeFilesWithoutEnFileName = fs
+        .readdirSync(localesI18nDirectory)
+        .filter(localeFile => localeFile.endsWith('.json') && localeFile !== 'en.json')
+
+    const enFileName = 'en.json'
+    const enFilePath = path.join(localesI18nDirectory, enFileName)
+    const enFileContent = JSON.parse(fs.readFileSync(enFilePath, 'utf-8'))
+
+    // Function to create a full key path
+    const createFullKey = (prefix, key) => prefix ? `${prefix}.${key}` : key
```

Comment:
personal preference: I would move these functions out so they're not nested. Since there's so many functions being defined and running inside `checkLocaleKeys` it makes it harder to track the function scope that's running